### DISCUSSION TITLE
Possibility to set different Start/End for timeline

### DISCUSSION THREAD

**Discussion by StefanZamecnik** [videojs/video.js#7749] [Q&A]
Hey,

does anyone know whether it's possible to set own start/end values for the players timeline and use not the one from the underlying video? We have a basic video file which can have subclips, but these subclips are just tc-in/out marks. We needed to set the start of the player to 0 but in this case, its not the start of the media, it's start of the subclip within the media.

Is this currenlty a possible scenario?

Thank you very much for your help.

Regards,
Stefan

**Comment by gkatsev** [comment]
There isn't anything built-in, but it should be possible to do something for this with [Video.js middleware](https://videojs.com/guides/middleware/)

**Comment by NickMoores** [comment]
@StefanZamecnik I'm actually looking to do exactly the same as you describe - how have you got on with the middleware? I'm about to give this a go.

**Comment by mwimmer-bcx** [comment]
Hey @NickMoores,

@StefanZamecnik and me did it like this:

We added a new custom button to our toolbar which is calling the following "applyInMarker" function:

```
window.applyInMarker = function () {
    if (player.rangePlaying)
        return;

    var time = player.currentTime();
    doLog('changed start-tc to ' + time + ' seconds');
    
    // apply InMarker
    const markers = player.markers.getMarkers();
    if (markers.length == 0) {
        player.markers.reset([{ time: time, text: 'In', role: 0 }, { time: -1, text: 'Out', role: 1 }]);
    } else {
        const outMarker = markers[1].role ? markers[1] : markers[0];
        player.markers.reset([{ time: time, text: 'In', role: 0 }, outMarker]);
    }
}
```

And we did the same for the second out marker with a second time entry.
If you need any further help for it, please let me know.
